### DISCUSSION TITLE
Help prevent "Benchy Hull Line"?

### DISCUSSION THREAD

**Discussion by GCheung55** [kennetek/gridfinity-rebuilt-openscad#244] [Q&A]
I'm seeing a line bulge in bins that corresponds with the top surface of the interior bottom, what I think is possibly known as the benchy hull line. This occurs with Orca Slicer.

![IMG_7074](https://github.com/user-attachments/assets/082eb1b5-3d0d-46e1-a738-a3df4399d95b)

From what I've [read online](https://help.prusa3d.com/article/the-benchy-hull-line_124745), the benchy hull line is "a sudden transition from sparse infill into full top layers".

How the benchy hull line appears starts with the top surface's internal bridge, which stretches across wall to wall (image below). As you might imagine, this is the "sudden transition from sparse to full top layers".

<img width="756" alt="image" src="https://github.com/user-attachments/assets/686efab0-f124-475b-9acd-5019fff88131">

From my experiments, the benchy hull line appears all around the bin for anything shorter than 4H, and only on the back and sides of the bin for anything 4H and greater. This is because the taller the bin, the bigger the scoop, which causes the top surface to be further away from the walls. Below is an image of a 1L x 2W x 5H bin. The internal bridge at the front is much further away from the front walls.

<img width="716" alt="image" src="https://github.com/user-attachments/assets/18ab0900-04ba-42af-a96c-1fc64dde40cd">

So, how can this be prevented? I'm not sure. I found [this post on reddit](https://www.reddit.com/r/prusa3d/comments/1bogicl/design_tips_on_eliminating_the_hull_line_effect/) that gives some hints that I'd like to try, but I'm not experienced enough with OpenSCAD to know what to do. 

What I've tried to do was create a void/space, with what looks like a gasket, behind the scoop and fillet. Below is an image of what the object looks like.

<img width="639" alt="image" src="https://github.com/user-attachments/assets/151258df-74bf-4069-af42-b1fb09c4c30c">

Then create the difference between the two and the outcome does stop the internal bridging from touching the walls. However the void I've created is much too far, causing other issues.

<img width="812" alt="image" src="https://github.com/user-attachments/assets/d951c7d0-d6fe-4712-ad81-bdaa2605eea1">

**TL;DR**
What I'd like to do is have the void fillet downwards to hug the internal space. What I'd want to try is copy the space of the bottom, create a slightly larger shell out of it, then use that shell to create the void. The purpose is to create a void that close to the  internal space so that the slicer and place the infill between the void and the inner/outer walls. Can someone guide me?

**Comment by GCheung55** [comment]
I also printed part of a bin without the top surface, exposing the sparse infill, to verify if the top surface was causing the issue. There is no benchy hull line where it would have been. 

![IMG_7075](https://github.com/user-attachments/assets/5ffba551-60f4-43f5-8b4d-4ae370183f0b)

**Reply by EmperorArthur** [reply]
There's some weirdness with scoops and cuts as is which needs to be fixed.

Just haven't gotten there yet.  I personally use Cura with zero scoop, and have not noticed anything.  However, I have not been looking either.

**Comment by GCheung55** [comment]
I created the void using Fusion360, since I'm more familiar with it. The void is .2mm, and just a vertical 1.4mm - no chamfer or fillets. I was able to use the void as a negative modifier and it does reduce majority of the bench hull line bulge. There are still a few lines that occasionally looks shifted that I'm not sure what is causing them, and I get a different issue with the layers having a different sheen on the layers compared to the others. The sheen issue looks to be a layer time issue.

## Layer Time Issue

More time is spent on the layers with top shells, in my case of 5 shells, which is somehow causing the outer walls on the outer surface to have a different sheen than the rest. This shouldn't be the case because I have Orca configured to not slow down the outer walls. When previewing, I can see that the outer walls of the outer surface on all expected layers have the same speed and flow. The only difference is layer time.

Speed Screenshot

<img width="813" alt="image" src="https://github.com/user-attachments/assets/6e5b51b4-43d1-4d15-8a47-f64325ec675b">

Flow Screenshot

<img width="689" alt="image" src="https://github.com/user-attachments/assets/bffae45b-c87f-41ff-b474-643f395f21df">

Layer Time Screenshot - (Red arrow points to the problematic layers)

![image](https://github.com/user-attachments/assets/4a67ca4b-9ae7-4551-b793-27d5ee6c9a8c)

## What's next?

I'm looking into modifying line order printing, because according to https://help.prusa3d.com/article/the-benchy-hull-line_124745, the devs modified the gcode to print the deck perimeters first, then the hull. I think that if I can print the bin's outer surface walls AFTER all the internal bridging, internal solid infill, and top surface are printed, then the sheen should also be addressed. The slicer's don't have this ability... yet. So I need to figure out gcode editing to fully confirm the fix.

In anycase, these are my findings thus far.
